src/helpers/batfish/navigation.js

```diff
@@ -93,10 +94,11 @@ function formatPages(siteBasePath, data, sections) {
     }));
 }
 
-function buildMultiLevels(sections, pages) {
+function buildMultiLevels(sections, pages, addPages) {
   return sections.reduce(
     (obj, section) => {
-      const sectionPages = pages.filter((f) => f.section === section.path);
+      let sectionPages = pages.filter((f) => f.section === section.path);
+      if (addPages) sectionPages = sectionPages.concat(addPages);
```

Comment:
```suggestion
      let sectionPages = pages.filter((f) => f.section === section.path);
      if (addPages) sectionPages = sectionPages.concat(section.addPages);
```

@danswick I think we still need to get `addPages` from the sections object, and not from the passed parameter.  I think our multi-structure test case is not catching this problem because we are only outputting one section (`maps`) and we should update it to use at least two sections.

For example, in iOS the site has many sections, let's just look at two:

```
[
      {
        path: 'maps',
        title: 'Maps SDK for iOS'
      },
      {
        path: 'navigation',
        title: 'Navigation SDK for Android'
      }
    ]
```

For each of these sections of the site, we want to add Tutorials and Troubleshooting links and each section will have its own product params appended to each link.

If we use `buildNavigation({siteBasePath, data, sections, addPages})` for iOS, how will `addPages` in this statement make sure each section adds the correct links to Tutorials and Troubleshooting? This code will append all `addPages` to every section.

This is why we created the `addPages` prop in `sections`  so that we can properly concat these new pages to the correct section.

Reply:
If we want to use `addPages` like is committed, then I think we could add an optional prop to the addPages shape (`section`) to help identify which section the links belongs to. Example:

```js
const addPages = [
  {
    title: 'Tutorials',
    path: 'https://docs.mapbox.com/help/tutorials?product=maps+sdk+for+ios',
    navOrder: 4,
    section: 'maps'
  },
  {
    title: 'Troubleshooting',
    path: 'https://docs.mapbox.com/help/troubleshooting?product=maps+sdk+for+ios',
    navOrder: 5,
    section: 'maps'
  },
  {
    title: 'Tutorials',
    path: 'https://docs.mapbox.com/help/tutorials?product=navigation+sdk+for+ios',
    navOrder: 3,
    section: 'navigation'
  },
  {
    title: 'Troubleshooting',
    path: 'https://docs.mapbox.com/help/troubleshooting?product=navigation+sdk+for+ios',
    navOrder: 4,
    section: 'navigation'
  }
];
```

And then we could filter `addPages` and then concat them to `sectionPages`:

```js
const thisSectionPages = addPages.filter(f => f.section === section.path);
if (thisSectionPages) sectionPages = sectionPages.concat(thisSectionPages);
```